src/System.Private.CoreLib/shared/System/Runtime/InteropServices/ArrayWithOffset.cs

```diff
@@ -4,19 +4,50 @@
 
 using System.Runtime.CompilerServices;
 
+#if BIT64
+using nuint = System.UInt64;
```

Comment:
`UIntPtr` would also work here, no?

Reply:
Not really because C# doesn't allow arithmetic operators on UIntPtr (CLR/IL does) which is the primary purpose here.